MekHQ/src/mekhq/campaign/personnel/marriage/AbstractMarriage.java

```diff
@@ -0,0 +1,297 @@
+        // marriages are limited to non-prisoner to non-prisoner marriages and prisoner to prisoner
+        // marriages.
+
+        if (person.equals(potentialSpouse)
+                || (canMarry(campaign, today, potentialSpouse, randomMarriage) != null)
+                || person.getGenealogy().checkMutualAncestors(potentialSpouse,
+                        campaign.getCampaignOptions().getCheckMutualAncestorsDepth())) {
+            return false;
+        } else if (randomMarriage) {
+            return person.getPrisonerStatus().isPrisoner() == potentialSpouse.getPrisonerStatus().isPrisoner();
+        } else {
+            return !potentialSpouse.getPrisonerStatus().isPrisoner() || person.getPrisonerStatus().isPrisoner();
+        }
+    }
+
+    /**
+     * This marries two people that are part of the same campaign together on the given date.
+     * @param campaign the campaign the two people are a part of
+     * @param today the current date
+     * @param origin the origin person being married
+     * @param spouse the person's spouse, which can be null if no marriage is to occur
+     * @param surnameStyle the style for how the two people's surnames will change as part of the marriage
+     */
+    public void marry(final Campaign campaign, final LocalDate today, final Person origin,
+                      final @Nullable Person spouse, final MergingSurnameStyle surnameStyle) {
+        if (spouse == null) {
+            return;
+        }
+
+        // Immediately set both Maiden Names, to avoid any divorce bugs (as the default is now an empty string)
+        origin.setMaidenName(origin.getSurname());
+        spouse.setMaidenName(spouse.getSurname());
+
+        // Then add them as spouses
+        origin.getGenealogy().setSpouse(spouse);
+        spouse.getGenealogy().setSpouse(origin);
+
+        // Apply the surname style changes
+        surnameStyle.apply(campaign, today, origin, spouse);
+
+        // Do the logging
+        PersonalLogger.marriage(origin, spouse, today);
+        PersonalLogger.marriage(spouse, origin, today);
+
+        campaign.addReport(String.format(resources.getString("marriage.report"), origin.getHyperlinkedName(),
+                spouse.getHyperlinkedName()));
+
+        // And finally we trigger person changed events
+        MekHQ.triggerEvent(new PersonChangedEvent(origin));
+        MekHQ.triggerEvent(new PersonChangedEvent(spouse));
+    }
+
+    //region New Day
+    /**
+     * Processes new day random marriage for an individual.
+     * @param campaign the campaign to process
+     * @param today the current day
+     * @param person the person to process
+     */
+    public void processNewDay(final Campaign campaign, final LocalDate today, final Person person) {
+        if (canMarry(campaign, today, person, true) != null) {
+            return;
+        }
+
+        if (randomOppositeSexMarriage(person)) {
+            marryRandomSpouse(campaign, today, person, false);
```

Comment:
This strangely reads as if marriages would happen each day for persons that are still free, but I suppose some low probability random check is happening elsewhere?

Reply:
randomOppositeSexMarriage is the method making the check for the person based on the specified time period (currently daily, but a future module may override it to be monthly). The current random method uses basic probability to do so. If it occurs, then the person will marry a random spouse.